### PR TITLE
Python: Add prompt caching support to Anthropic connector

### PR DESCRIPTION
**Pull request by Vizhy**
## Summary

Adds opt-in prompt caching support to the Python Anthropic connector via a new `AnthropicCacheSettings` model and a `prepare_settings_dict()` override that injects Anthropic `cache_control` blocks into the outbound request payload.

**What's included:**
- `AnthropicCacheSettings` — inherits `KernelBaseSettings` with `env_prefix = ANTHROPIC_CACHE_`. Caching can be toggled via environment variables (`ANTHROPIC_CACHE_ENABLED`, `ANTHROPIC_CACHE_INCLUDE_SYSTEM`, `ANTHROPIC_CACHE_INCLUDE_TOOLS`, `ANTHROPIC_CACHE_TTL`) or set explicitly in code.
- `AnthropicChatPromptExecutionSettings.cache` field (excluded from serialization) + `prepare_settings_dict()` override that injects `cache_control` on the system message and/or the last tool definition.
- `AnthropicCacheSettings` exported from `semantic_kernel.connectors.ai.anthropic`.
- Unit tests covering all classmethods, both TTLs, all injection combinations, edge cases, env-var loading, and the no-overwrite guard.
- Sample: `samples/concepts/caching/anthropic_prompt_caching.py`.

**Copilot review addressed (commit f71799f4d):**
- Inherits `KernelBaseSettings` (not bare `BaseModel`) — consistent with rest of SDK; enables env-var support.
- Renamed `cache_system`/`cache_tools` → `include_system`/`include_tools` — removes redundant cache prefix.
- Replaced `copy.deepcopy` with shallow list + dict spread — cheaper for large tool catalogs.
- Injection skips if `cache_control` already present — avoids clobbering caller's explicit setting.
- TTL tests moved to `prepare_settings_dict()` public surface (no longer calling private `_cache_control()`).

**Cache TTL:**
- `5m` (default): 1.25x write cost, 0.1x read cost. Breaks even after one cache hit.
- `1h`: 2x write cost, 0.1x read cost. Breaks even after two cache hits.

**Usage:**
```python
from semantic_kernel.connectors.ai.anthropic import AnthropicCacheSettings

settings = AnthropicChatPromptExecutionSettings(
    cache=AnthropicCacheSettings.on(ttl=1h),
)
# or via env: ANTHROPIC_CACHE_ENABLED=true ANTHROPIC_CACHE_TTL=1h
```

### CODE REVIEW

**Comment by Vizhy** [comment]
@microsoft-github-policy-service agree

**Comment by Vizhy** [comment]
Thanks for the thorough automated review — two valid issues were caught and both are addressed in the follow-up commit (da6de6422):

**1. TTL value fix (Correctness)**
`_cache_control()` now emits `{"ttl": "1h"}` (string) instead of `{"ttl": 3600}` (integer), correctly matching the `CacheControlEphemeralParam` SDK type definition. The corresponding test assertions have been updated to match.

**2. Pre-structured system blocks (Design Approach)**
`prepare_settings_dict()` now handles both input shapes for `system`:
- `str` → wrapped into a single content block with `cache_control`
- `list[dict]` → `cache_control` injected on the last block (same pattern used for tools), with a `copy.deepcopy` to avoid mutation

A test covering the `list[dict]` case has been added to document this behaviour explicitly.

File: python/semantic_kernel/connectors/ai/anthropic/prompt_execution_settings/anthropic_prompt_execution_settings.py [hunk 1]
```diff
@@ -12,6 +13,88 @@
 logger = logging.getLogger(__name__)
 
 
+class AnthropicCacheSettings(BaseModel):
+    """Configuration for Anthropic prompt caching.
+
+    Controls which parts of the request receive cache_control injection.
+
```
[the remark this answers is not in the document]
Comment (Vizhy) [reply python/semantic_kernel/connectors/ai/anthropic/prompt_execution_settings/anthropic_prompt_execution_settings.py]: Fixed in f71799f4d. `AnthropicCacheSettings` now inherits `KernelBaseSettings` with `env_prefix = ANTHROPIC_CACHE_`, consistent with the rest of the SDK (`validate_assignment`, `populate_by_name`, `arbitrary_types_allowed`). This also unlocks env-var control out of the box: `ANTHROPIC_CACHE_ENABLED=true`, `ANTHROPIC_CACHE_INCLUDE_SYSTEM=true`, `ANTHROPIC_CACHE_TTL=1h`, etc. Took the opportunity to rename `cache_system`/`cache_tools` to `include_system`/`include_tools` to remove the redundant cache prefix on fields inside a cache settings class.

File: python/semantic_kernel/connectors/ai/anthropic/prompt_execution_settings/anthropic_prompt_execution_settings.py [hunk 2]
```diff
@@ -53,3 +143,26 @@ def validate_tool_choice(self) -> "AnthropicChatPromptExecutionSettings":
...
+        if self.cache.cache_system:
+            system = data.get("system")
+            if isinstance(system, str) and system:
+                data["system"] = [{"type": "text", "text": system, "cache_control": cache_control}]
+
+        if self.cache.cache_tools:
+            tools: list[dict[str, Any]] | None = data.get("tools")
+            if tools:
+                tools = copy.deepcopy(tools)
+                tools[-1]["cache_control"] = cache_control
```
[the remark this answers is not in the document]
Comment (Vizhy) [reply python/semantic_kernel/connectors/ai/anthropic/prompt_execution_settings/anthropic_prompt_execution_settings.py]: Fixed in f71799f4d. Injection now checks `cache_control not in tools[-1]` (and the same for system blocks) before writing — existing values are preserved as-is.

File: python/tests/unit/connectors/ai/anthropic/test_anthropic_request_settings.py [hunk 3]
```diff
@@ -127,3 +128,172 @@ def test_tool_choice_none():
...
+
+
+def test_cache_control_5m():
+    ctrl = AnthropicCacheSettings.on(ttl="5m")._cache_control()
+    assert ctrl == {"type": "ephemeral"}
+
+
+def test_cache_control_1h():
+    ctrl = AnthropicCacheSettings.on(ttl="1h")._cache_control()
+    assert ctrl == {"type": "ephemeral", "ttl": 3600}
```
[the remark this answers is not in the document]
Comment (Vizhy) [reply python/tests/unit/connectors/ai/anthropic/test_anthropic_request_settings.py]: Fixed in f71799f4d. Replaced the two `_cache_control()` direct tests (`test_cache_control_5m` / `test_cache_control_1h`) with `prepare_settings_dict()` equivalents that validate the same TTL output via the public API. The private helper is now only exercised transitively.

File: python/semantic_kernel/connectors/ai/anthropic/prompt_execution_settings/anthropic_prompt_execution_settings.py [hunk 4]
```diff
@@ -53,3 +143,26 @@ def validate_tool_choice(self) -> "AnthropicChatPromptExecutionSettings":
...
+                data["tools"] = tools
```
[the remark this answers is not in the document]
Comment (Vizhy) [reply python/semantic_kernel/connectors/ai/anthropic/prompt_execution_settings/anthropic_prompt_execution_settings.py]: Fixed in f71799f4d. Replaced `copy.deepcopy` with a shallow list + dict spread: `[*tools[:-1], {**tools[-1], cache_control: cache_control}]`. Only the last element is modified so only that dict needs copying — no full deep clone of the entire tools list.

**Comment by Vizhy** [comment]
@moonbox3
